src/executer.ts

```diff
@@ -0,0 +1,57 @@
+import { TaskQueue } from './types'
+
+const QUEUE = 'commandQueue'
+const AGENTS = 'agents'
+
+export default class Executer {
+  private createAgent: Function
+  private commandQueue: any[]
+  private agents: any
+
+  constructor (Agent: any) {
+    this.createAgent = () => new Agent()
```

Comment:
why wrap create instance ?

Reply:
Because, typescript do not allow `new this.agent()` statement.

> TS2351: Cannot use 'new' with an expression whose type lacks a call or construct signature.